runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -274,21 +274,40 @@ void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace
      *                         if null, only future updates will be returned
      */
     public void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace,
-                   @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
-                   @Nullable Timestamp timestamp) {
-        runtime.getTableRegistry().getStreamManager()
+                          @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
+                          @Nullable Timestamp timestamp) {
+        runtime.getTableRegistry().getStreamingManager()
                 .subscribe(streamListener, namespace, streamTag, tablesOfInterest,
                         (timestamp == null) ? getTimestamp().getSequence() : timestamp.getSequence());
     }
 
+    /**
+     * Subscribe to transaction updates on specific tables with the streamTag in the namespace.
+     * Objects returned will honor transactional boundaries.
+     *
+     * @param streamListener   client listener for callback
+     * @param namespace        the CorfuStore namespace to subscribe to
+     * @param streamTag        only updates of tables with the stream tag will be polled
+     * @param tablesOfInterest only updates from these tables of interest will be sent to listener
+     * @param timestamp        if specified, all stream updates after this timestamp will be returned
+     *                         if null, only future updates will be returned
+     * @param bufferSize       maximum size of buffered transaction entries
+     */
+    public void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace,
+                          @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
+                          @Nullable Timestamp timestamp, int bufferSize) {
+        runtime.getTableRegistry().getStreamingManager()
+                .subscribe(streamListener, namespace, streamTag, tablesOfInterest,
+                        (timestamp == null) ? getTimestamp().getSequence() : timestamp.getSequence(), bufferSize);
+    }
+
     /**
      * Gracefully shutdown a streamer.
      * Once this call returns no further stream updates will be returned.
      *
      * @param streamListener - callback context.
      */
     public void unsubscribe(@Nonnull StreamListener streamListener) {
-        runtime.getTableRegistry().getStreamManager()
-                .unsubscribe(streamListener);
+        runtime.getTableRegistry().getStreamingManager().unsubscribe(streamListener);
```

Comment:
This looks like a leak. You are allowing subscribe to be called from `StreamManager` and `StreamingManager` but `unsubscribe` only invokes `StreamingManager::unsubscribe`

Reply:
There is not a lot of dependencies on `StreamManager`, I would suggest removing it instead of deprecating it. Its cleaner than just assuming that API won't be called.

Reply:
Done, thanks for finding out, I renamed the new subscribe/unscribe to subscribeListener/unsubscribeListener, which is clearer name and deperated the old unscribe API. 

The reason is we don't want to break the interface, some verticals are using the old subscribe api, and to convert to new API we need them to clean up their code, which is better to be in a separate effort to unblock merging process.